### PR TITLE
Disallow unwrap being used in hivesim-rs

### PR DESCRIPTION
**Pull request by KolbyML**
unwrap makes panics harder to debug, this change will allow for easier diagnostics

### CODE REVIEW

File: hivesim-rs/src/simulation.rs [hunk 1]
```diff
@@ -66,21 +66,23 @@ impl Simulation {
...
-            .json(&body)
-            .send()
-            .await
-            .unwrap()
-            .json::<SuiteID>()
-            .await
-            .unwrap()
+
+        match client.post(url).json(&body).send().await {
+            Ok(response) => match response.json::<SuiteID>().await {
```
Comment (ogenev) [inline hivesim-rs/src/simulation.rs:71]: Can we leverage `expect()` here?

**Review by ogenev — COMMENTED** [review]
I think we can leverage [expect](https://doc.rust-lang.org/std/result/enum.Result.html#method.expect) in most cases, instead of matching the error and panicking.

**Comment by KolbyML** [comment]
Ok I resolved the concerns and switched them to expect. Thank you for letting me know that. I didn't know expect did the "x: y" format I wanted :).

**Review by carver — APPROVED** [review]
Yup, definitely an improvement!

My strategy is to avoid having the same error string anywhere twice in the codebase. It makes hunting down unfamiliar bugs so fast, because you immediately know the line even if something funny happens with not getting the right line number of the panic.

Most of these errors look unique, which is great, but a couple aren't. See if you can naturally update the "Failed to convert response to json" lines. One example would be something like "Failed to convert suite start response to json".

**Comment by KolbyML** [comment]
That is a very good suggestion :D thank you for the help. I resolved that concern so I will merge it after CI is done

**Pull request merged**